src/v/datalake/record_multiplexer.cc

```diff
@@ -268,8 +288,22 @@ record_multiplexer::operator()(model::record_batch batch) {
     co_return ss::stop_iteration::no;
 }
 
+ss::future<writer_error> record_multiplexer::flush_writers() {
+    if (_error) {
+        co_return *_error;
+    }
+    auto result = co_await ss::coroutine::as_future(ss::max_concurrent_for_each(
+      _writers, 10, [](auto& entry) { return entry.second->flush(); }));
```

Comment:
Similar comment to the other one regarding the `10` value. Why and when we should consider changing it?